fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -233,6 +233,19 @@ private boolean insertField(LuceneDocumentFromRecord.DocumentField field, final
         return suggestionAdded;
     }
 
+    private FieldType getTextFieldType(boolean isStored) {
+        FieldType ft = new FieldType();
+        ft.setIndexOptions(LuceneRecordContextProperties.getIndexOptions(this.state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_FULL_TEXT_FIELD_INDEX_OPTIONS), LOG));
+        ft.setTokenized(true);
+        ft.setStored(isStored);
+        ft.setStoreTermVectors(this.state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_FULL_TEXT_WITH_TERM_VECTORS));
+        ft.setStoreTermVectorPositions(this.state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_FULL_TEXT_WITH_TERM_VECTOR_POSITIONS));
+        ft.setOmitNorms(this.state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_FULL_TEXT_WITH_OMIT_NORMS));
```

Comment:
This would seem to suggest that we're saying that all fields get the same value for these values. Is that the idea, or are there use cases for having different values for these options for different fields?

Reply:
Different values per field.  For example, we do not want to store positions for specific fields.